nominatim/tools/special_phrases.py

```diff
@@ -0,0 +1,259 @@
+"""
+    Functions to import special phrases into the database.
+"""
+import logging
+import os
+import re
+import subprocess
+import json
+from os.path import isfile
+from icu import Transliterator
+from psycopg2.sql import Identifier, Literal, SQL
+from nominatim.tools.exec_utils import get_url
+
+LOG = logging.getLogger()
+class SpecialPhrasesImporter():
+    # pylint: disable-msg=too-many-instance-attributes
+    # pylint: disable-msg=too-few-public-methods
+    """
+        Class handling the process of special phrases importations.
+    """
+    def __init__(self, config, phplib_dir, db_connection) -> None:
+        self.db_connection = db_connection
+        self.config = config
+        self.phplib_dir = phplib_dir
+        self.black_list, self.white_list = self._load_white_and_black_lists()
+        #Compile the regex here to increase performances.
+        self.occurence_pattern = re.compile(
+            r'\| ([^\|]+) \|\| ([^\|]+) \|\| ([^\|]+) \|\| ([^\|]+) \|\| ([\-YN])'
+        )
+        self.sanity_check_pattern = re.compile(r'^\w+$')
+        self.transliterator = Transliterator.createFromRules("special-phrases normalizer",
+                                                             self.config.TERM_NORMALIZATION)
+
+    def import_from_wiki(self, languages=None):
+        """
+            Iterate through all specified languages and
+            extract corresponding special phrases from the wiki.
+        """
+        #Get all languages to process.
+        languages = self._load_languages() if not languages else languages
+
+        #array for pairs of class/type
+        class_type_pairs = set()
```

Comment:
Something is very wrong now. This set is never written, so I suspect line 50 does nothing? You might need a test for the full function of import_from_wiki() as well.

NB: the comment in the line above is wrong. This is not an array.

Reply:
Yes I am doing big mistakes there I should have tested my code more. By doing it too fast I am not transferring the set throughout functions...